src/extensions/common/ClientSidePlot.tsx

```diff
@@ -113,9 +113,11 @@ const ClientSidePlot: FunctionComponent<{
                 </Box>
             </Box>
         );
-    } else if (calculationStatus === 'finished') {
+    } else if ((calculationStatus === 'finished') && (plotData)) {
         // TODO: Follow-up on distinction b/w this and <PlotComponent arg1={} arg2={} ... />
-        return plotComponent({boxSize, plotData, argsObject: plotComponentArgs, title});
+        return <PlotComponent
```

Comment:
This is the proper way to embed the PlotComponent. The previous way sort of worked, but it turned out it did not handle hooks properly. If you notice in the previous lines, we were evaluating the function every time render was called, and so the hooks were being executed in the ClientSidePlot render rather than in the PlotComponent render.